library/alloc/src/collections/vec_deque/into_iter.rs

```diff
@@ -36,6 +36,18 @@ impl<T> Iterator for IntoIter<T> {
         let len = self.inner.len();
         (len, Some(len))
     }
+
+    #[inline]
+    unsafe fn __iterator_get_unchecked(&mut self, idx: usize) -> Self::Item
+    where
+        Self: TrustedRandomAccess,
+    {
+        // Safety: Callers are only allowed to pass an index that is in bounds
```

Comment:
And maybe also add a comment why it's safe to read the value (because constrained to `Item: Copy` and the API contract of `TrustedRandomAccess` requires the same index to not be accessed again)

Reply:
Since the items are `Copy` it wouldn't even matter if a value is ready multiple times.

Reply:
Yeah, still worth mentioning in case someone wants to lift the `Copy` bound later